auth/auth.ts

```diff
@@ -105,19 +106,25 @@ export class MockAuth implements firebase.auth.Auth {
 
   private signIn(
     user: User,
-    additionalUserInfo: firebase.auth.AdditionalUserInfo | null = null
+    { isNewUser }: { isNewUser: boolean }
   ): Promise<firebase.auth.UserCredential> {
     this.currentUser = user;
     this.authStateEvents.forEach((listener) => {
       listener(user);
     });
 
-    return Promise.resolve<firebase.auth.UserCredential>({
-      user,
-      additionalUserInfo,
-      credential: null,
-      operationType: "signIn",
-    });
+    return Promise.resolve(new UserCredential(user, "signIn", isNewUser));
+  }
+
+  private async signInWithSocial(provider: firebase.auth.AuthProvider) {
+    const data = await this.store.consumeSocialMock(provider);
+    let user = this.store.findByProviderAndEmail(data.email, provider.providerId);
+    if (user) {
+      return this.signIn(user, { isNewUser: false });
+    }
+
+    user = this.store.add({ ...data, providerId: provider.providerId });
+    return this.signIn(user, { isNewUser: true });
```

Comment:
Little confused why it appears that we are removing the data from the store just to add it again

Reply:
Can you elaborate? I'm not following.

Reply:
Well if I am not mistaken `consumeSocialMock()` removes the provider from the store, but `this.store.add()` would add it back to the store.

Maybe `consumeSocialMock()` (and other functions) could use a JSDoc comment to elaborate on what it does to make this more clear?

Reply:
Added some comments around, let me know if it's clearer now!